tests/components/hunterdouglas_powerview/test_config_flow.py

```diff
@@ -260,97 +131,105 @@ async def test_form_homekit_and_dhcp(
         context={"source": source},
         data=discovery_info,
     )
-    assert result3["type"] == "abort"
+    assert result3["type"] == FlowResultType.ABORT
 
 
-async def test_discovered_by_homekit_and_dhcp(hass: HomeAssistant) -> None:
+@pytest.mark.usefixtures("mock_setup_entry", "mock_hunterdouglas_full")
+@pytest.mark.parametrize(
+    ("homekit_source", "homekit_discovery", "api_version"), HOMEKIT_DATA
+)
+@pytest.mark.parametrize(
+    ("dhcp_source", "dhcp_discovery", "dhcp_api_version"), DHCP_DATA
+)
+async def test_discovered_by_homekit_and_dhcp(
+    hass: HomeAssistant,
+    homekit_source: str,
+    homekit_discovery: zeroconf.ZeroconfServiceInfo,
+    api_version: str,
+    dhcp_source: str,
+    dhcp_discovery: dhcp.DhcpServiceInfo,
+    dhcp_api_version: str,
+) -> None:
     """Test we get the form with homekit and abort for dhcp source when we get both."""
 
-    mock_powerview_userdata = _get_mock_powerview_userdata()
-    with patch(
-        "homeassistant.components.hunterdouglas_powerview.UserData",
-        return_value=mock_powerview_userdata,
-    ):
-        result = await hass.config_entries.flow.async_init(
-            DOMAIN,
-            context={"source": config_entries.SOURCE_HOMEKIT},
-            data=HOMEKIT_DISCOVERY_INFO,
-        )
+    if api_version != dhcp_api_version:
+        return
```

Comment:
We want to avoid branching inside tests as it makes them harder to maintain and troubleshoot.

Please use pytest skipif or split the test into two tests